lib/pdf-editor-view.js

```diff
@@ -490,6 +495,32 @@ export default class PdfEditorView extends ScrollView {
     }, () => this.finishUpdate());
   }
 
+  computeMaxPageWidth(){
+    Promise.all(
+      _.range(1, this.pdfDocument.numPages + 1).map((pdfPageNumber) =>
+        this.pdfDocument.getPage(pdfPageNumber).then((pdfPage) =>
+          pdfPage.getViewport(1.0).width
+        )
+      )
+    ).then((pdfPageWidths) => {
+      this.maxPageWidth = Math.max(...pdfPageWidths);
+      this.zoomFit();
+      // console.log('Max width: ' + this.maxPageWidth);
+    })
+  }
+
+  zoomFit() {
+    if (this.maxPageWidth == 0) {
+      this.computeMaxPageWidth();
+      return;
+    }
+    let fitScale = this[0].clientWidth / this.maxPageWidth;
```

Comment:
If the PDF is auto-reloaded due to it being updated, would the maxPageWidth remain set to the max page width of the PDF's old version? In other words, after a PDF if auto-reloaded due to an update on disk, do we need to set maxPageWidth to 0 to force a re-computation?

Reply:
Done.